### PR TITLE
feat(helm): update helm search

### LINKED ISSUE #1226: ref: helm search should use the chart indices, instead of walking dirs.
Search predated the index files, and so used a walker to find charts. It should use index files now.

See https://github.com/kubernetes/helm/blob/master/cmd/helm/search.go#L48

### PR DESCRIPTION
**Pull request by technosophos**
Switch 'helm search' from file crawling to using the indices. Also
add scorable indexing, forward porting the search code I originally
wrote for Helm Classic.

Closes #1226
Partially addresses #1199

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/helm/1229)

### CODE REVIEW

**Comment by technosophos** [comment]
To test:

- Try out `helm search` in the usual way
- Give regexps a shot with `helm search -r 'alp[ine]*'` or something similar. Depending on your shell, you may need to single- or double-quote your regexp

File: cmd/helm/search/search_test.go [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+			ex := tt.expect[i]
+			if got.Name != ex.Name {
+				t.Errorf("%s[%d]: Expected name %q, got %q", tt.name, i, ex.Name, got.Name)
+			}
+		}
+
+	}
+}
+
+/*
```
Comment (adamreese) [inline cmd/helm/search/search_test.go]: is the commented section here for a reason?

**Comment by adamreese** [comment]
manually tested for awesomeness

File: cmd/helm/search/search_test.go (hunk 1 above)
Comment (technosophos) [inline cmd/helm/search/search_test.go]: Nope. That was left over from before I table-ized the tests. I'll remove them.

**Pull request merged**